src/iceberg/manifest_reader.h

```diff
@@ -35,35 +35,32 @@ namespace iceberg {
 class ICEBERG_EXPORT ManifestReader {
  public:
   virtual ~ManifestReader() = default;
-  virtual Result<std::span<std::unique_ptr<ManifestEntry>>> Entries() const = 0;
+  virtual Result<std::vector<ManifestEntry>> Entries() const = 0;
 
- private:
-  std::unique_ptr<Reader> reader_;
+  static Result<std::unique_ptr<ManifestReader>> MakeReader(
+      const std::string& manifest_location, std::shared_ptr<FileIO> file_io,
+      std::shared_ptr<Schema> partition_schema);
 };
 
 /// \brief Read manifest files from a manifest list file.
 class ICEBERG_EXPORT ManifestListReader {
  public:
   virtual ~ManifestListReader() = default;
-  virtual Result<std::span<std::unique_ptr<ManifestFile>>> Files() const = 0;
+  virtual Result<std::vector<ManifestFile>> Files() const = 0;
 
- private:
-  std::unique_ptr<Reader> reader_;
+  static Result<std::unique_ptr<ManifestListReader>> MakeReader(
+      const std::string& manifest_list_location, std::shared_ptr<FileIO> file_io);
 };
 
 /// \brief Creates a reader for the manifest list.
 /// \param file_path Path to the manifest list file.
 /// \return A Result containing the reader or an error.
 Result<std::unique_ptr<ManifestListReader>> CreateManifestListReader(
-    std::string_view file_path) {
-  return NotImplemented("CreateManifestListReader is not implemented yet.");
-}
+    std::string_view file_path);
```

Comment:
We need to remove `CreateManifestListReader` and `CreateManifestReader`, then use above static functions in the table_scan.cc as they are used there.